tests/parsing/test_full_telemetry_parsing.py

```diff
@@ -1,3 +1,33 @@
-# import pytest
+import pytest
+from modules.telemetry.telemetry_utils import parse_rn2483_transmission, parse_radio_block, is_valid_packet_header, ParsedTransmission
+from modules.telemetry.v1.block import PacketHeader
+from modules.misc.config import load_config
 
-# from modules.misc.config import load_config
+
+config = load_config("config.json")
+# Fixtures 
+
+@pytest.fixture
+def approved_callsigns() -> dict[str, str]:
+    return config.approved_callsigns
+
+@pytest.fixture
+def valid_packet_header() -> PacketHeader:
+    # first 32 characters of a packet
+    hdr = "564133494e490000000c010137000000"
+    pkt_hdr = PacketHeader.from_hex(hdr)
+    return pkt_hdr
+@pytest.fixture
+def non_valid_packet() -> PacketHeader:
+    # changed e in valid pakct header fixture to a 2
```

Comment:
This just changes the call sign from VA3INI to VA3IBI.
I'd recommend adding other tests (such as version 0, length 0, bad block number, all zeroes, etc)

Mainly as just callsign seems like a silly check as you could just add VA3IBI into the approved callsigns list and break this test.

Reply:
Seconding this!

Reply:
One thing I will say is that the function under test only checks two things:
- Is the call sign recognized?
- Is the version supported?

This function should probably be changed to `from_recognized_callsign` and only report if the call sign is recognized. The version checking should take place when the PacketHeader is constructed and raise an exception if the version isn't supported.